From 24ebbdff35fc6236bbfc5a970a4a808945c1a2d8 Mon Sep 17 00:00:00 2001
From: Eric Fourmont <eric.fourmont-ext@st.com>
Date: Mon, 10 Jul 2023 11:38:42 +0200
Subject: [PATCH 0362/1141] arm64: dts: st: use rif-open resmem configuration
 on stm32mp257f-ev1

Define 256MB region for OSPI1 instead of OSPI2.
Discard reserved memory for PCIE, VDEC and VENC.
Move GPU reserved memory just before reserved memory for LTDC.

Signed-off-by: Loic PALLARDY <loic.pallardy@st.com>
Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Change-Id: I5182de99b112ff90ffcc096783e2254db5674570
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/316648
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 .../st/stm32mp257f-ev1-ca35tdcid-resmem.dtsi  | 29 +++++++------------
 1 file changed, 11 insertions(+), 18 deletions(-)

diff --git a/arch/arm64/boot/dts/st/stm32mp257f-ev1-ca35tdcid-resmem.dtsi b/arch/arm64/boot/dts/st/stm32mp257f-ev1-ca35tdcid-resmem.dtsi
index 5917de4ce494..7f3e1ce76e9a 100644
--- a/arch/arm64/boot/dts/st/stm32mp257f-ev1-ca35tdcid-resmem.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp257f-ev1-ca35tdcid-resmem.dtsi
@@ -6,8 +6,8 @@
 
 /*
  * STM32MP25 reserved memory device tree configuration
- * Project : eco
- * Generated by XLmx tool version 2.2 - 6/1/2023 2:34:56 PM
+ * Project : open
+ * Generated by XLmx tool version 2.2 - 7/4/2023 9:06:24 AM
  */
 
 / {
@@ -80,48 +80,56 @@ tfm_its: tfm-its@42001000 {
 		};
 
 		/* Octo Memory Manager reserved memory declaration */
-		mm_ospi2: mm-ospi@60000000 {
+		mm_ospi1: mm-ospi@60000000 {
 			reg = <0x0 0x60000000 0x0 0x10000000>;
 			no-map;
 		};
 
 		/* DDR reserved memory declaration */
 		tfm_code: tfm-code@80000000 {
+			compatible = "shared-dma-pool";
 			reg = <0x0 0x80000000 0x0 0x100000>;
 			no-map;
 		};
 
 		cm33_cube_fw: cm33-cube-fw@80100000 {
+			compatible = "shared-dma-pool";
 			reg = <0x0 0x80100000 0x0 0x800000>;
 			no-map;
 		};
 
 		tfm_data: tfm-data@80900000 {
+			compatible = "shared-dma-pool";
 			reg = <0x0 0x80900000 0x0 0x100000>;
 			no-map;
 		};
 
 		cm33_cube_data: cm33-cube-data@80a00000 {
+			compatible = "shared-dma-pool";
 			reg = <0x0 0x80a00000 0x0 0x800000>;
 			no-map;
 		};
 
 		ipc_shmem_1: ipc-shmem-1@81200000 {
+			compatible = "shared-dma-pool";
 			reg = <0x0 0x81200000 0x0 0xf8000>;
 			no-map;
 		};
 
 		vdev0vring0: vdev0vring0@812f8000 {
+			compatible = "shared-dma-pool";
 			reg = <0x0 0x812f8000 0x0 0x1000>;
 			no-map;
 		};
 
 		vdev0vring1: vdev0vring1@812f9000 {
+			compatible = "shared-dma-pool";
 			reg = <0x0 0x812f9000 0x0 0x1000>;
 			no-map;
 		};
 
 		vdev0buffer: vdev0buffer@812fa000 {
+			compatible = "shared-dma-pool";
 			reg = <0x0 0x812fa000 0x0 0x6000>;
 			no-map;
 		};
@@ -146,21 +154,6 @@ gpu_reserved: gpu-reserved@fa800000 {
 			no-map;
 		};
 
-		pcie_reserved: pcie-reserved@fd800000 {
-			reg = <0x0 0xfd800000 0x0 0x800000>;
-			no-map;
-		};
-
-		vdec_reserved: vdec-reserved@fe000000 {
-			reg = <0x0 0xfe000000 0x0 0x800000>;
-			no-map;
-		};
-
-		venc_reserved: venc-reserved@fe800000 {
-			reg = <0x0 0xfe800000 0x0 0x800000>;
-			no-map;
-		};
-
 		ltdc_sec_layer: ltdc-sec-layer@fe800000 {
 			reg = <0x0 0xfe800000 0x0 0x800000>;
 			no-map;
-- 
2.39.2

